src/mds/objects.py

```diff
@@ -227,6 +226,65 @@ async def create_object_for_id(
     return JSONResponse(response, HTTP_201_CREATED)
 
 
+@mod.get("/objects/{guid:path}/download")
+async def get_object_signed_download_url(
+    guid: str,
+    request: Request,
+) -> JSONResponse:
+    """
+    Send a GET request to the data access service to generate a signed download
+    url for the given GUID or alias. Returns the generated signed download url
+    to the user.
+
+    Args:
+        guid (str): indexd GUID or alias
+        request (Request): starlette request (which contains reference to FastAPI app)
+
+    Returns:
+        200: { "guid": GUID or alias, "download_url": signed download url }
+        404: if the data access service can not find GUID/alias in indexd
+        403: if the data access service returns a 401 or a 403
+        500: if there is an error making the request to the data access service
+        or the data access service returns any other 400-range or 500-range
+        error
+    """
+    try:
+        endpoint = (
+            config.DATA_ACCESS_SERVICE_ENDPOINT.rstrip("/") + f"/data/download/{guid}"
+        )
+        auth_header = str(request.headers.get("Authorization", ""))
+        logger.debug(
+            f"Attempting to GET signed download url from data access service for GUID or alias '{guid}'"
+        )
+        response = await request.app.async_client.get(
+            endpoint, headers={"Authorization": auth_header}
+        )
+        response.raise_for_status()
+        signed_download_url = response.json().get("url")
+    except httpx.HTTPError as err:
+        msg = f"Unable to get signed download url from data access service for GUID or alias '{guid}'"
+        logger.error(f"{msg}\nException:\n{err}", exc_info=True)
+        if err.response:
+            logger.error(
+                f"data access service response status code: {err.response.status_code}\n"
+                f"data access service response text:\n{getattr(err.response, 'text')}"
+            )
+            if err.response.status_code in (401, 403):
+                raise HTTPException(
+                    HTTP_403_FORBIDDEN,
+                    f"{msg}. You do not have access to generate the download url for GUID or alias '{guid}'.",
+                )
+            elif err.response.status_code == 404:
+                raise HTTPException(
+                    HTTP_404_NOT_FOUND,
+                    f"{msg}. Record with GUID or alias '{guid}' was not found in indexd.",
+                )
+        raise HTTPException(HTTP_500_INTERNAL_SERVER_ERROR, msg)
+
+    response = {"guid": guid, "download_url": signed_download_url}
```

Comment:
I don't think we need to return guid here, and let's maybe simplify and just use `url`? I assume you're trying to be consistent b/c the upload is "upload_url" but I think in this situation b/c the URL explicitly has "download", using `url` is a little cleaner? that's a personal opinion and sorta a nit-pick, take it or leave it